medical-portal/src/API/RSBC.DMF.MedicalPortal.API/Controllers/ChefsController.cs

```diff
@@ -140,6 +141,38 @@ public async Task<ActionResult> PutSubmission([FromQuery] string caseId, [FromBo
             }
             else if (status == SubmissionStatus.Final)
             {
+                Dictionary<string, object> chefsFlags = submission.Flags;
+
+                // get a list of all available Case Flags 
+                var f = cmsAdapterClient.GetAllFlags(new EmptyRequest());
+                if (f == null || f.Flags.Count == 0)
+                {
+                    logger.LogInformation("Could not find all flags in the CMS");
+                    return StatusCode((int)HttpStatusCode.NotFound,
+                        "Not found error - could not find all flags in the CMS");
+                }
+
+                IEnumerable<Flag> allCaseFlags = mapper.Map<IEnumerable<Flag>>(f.Flags);
+                
+                // if any Case Flags are present and active (true) in CHEFS, update Case and set IsCleanPass to false
+                var matchedFlags = allCaseFlags.Where(flag =>
```

Comment:
for next time, you can shorten this slightly by using Linq.Any, it will return a bool if condition is met